src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+        bytes32 _winningForkPayoutDistributionHash = _currentBranch.getForkingMarket().getFinalPayoutDistributionHash();
+        Branch _destinationBranch = _currentBranch.getChildBranch(_winningForkPayoutDistributionHash);
+        ReportingWindow _newReportingWindow = _destinationBranch.getNextReportingWindow();
+        _newReportingWindow.migrateMarketInFromNibling();
+        reportingWindow.removeMarket();
+        reportingWindow = _newReportingWindow;
+        // reset to unreported state
+        limitedReportersDisputeBondToken = DisputeBondToken(0);
+        allReportersDisputeBondToken = DisputeBondToken(0);
+        tentativeWinningPayoutDistributionHash = 0;
+        return true;
+    }
+
+    ////////
+    //////// Helpers
+    ////////
+
+    function getReportingToken(uint256[] _payoutNumerators) public returns (ReportingToken) {
+        bytes32 _payoutDistributionHash = derivePayoutDistributionHash(_payoutNumerators);
+        ReportingToken _reportingToken = reportingTokens[_payoutDistributionHash];
+        if (address(_reportingToken) == NULL_ADDRESS) {
+            _reportingToken = ReportingTokenFactory(controller.lookup("ReportingTokenFactory")).createReportingToken(controller, this, _payoutNumerators);
+            reportingTokens[_payoutDistributionHash] = _reportingToken;
+        }
+        return _reportingToken;
+    }
+
+    function fundDisputeBondWithReputation(address _bondHolder, DisputeBondToken _disputeBondToken, uint256 _bondAmount) private returns (bool) {
+        require(_bondHolder == _disputeBondToken.getBondHolder());
+        reportingWindow.getReputationToken().trustedTransfer(_bondHolder, _disputeBondToken, _bondAmount);
+        return true;
+    }
+
+    function transferIncorrectDisputeBondsToWinningReportingToken() private returns (bool) {
+        require(isFinalized());
+        ReputationToken _reputationToken = reportingWindow.getReputationToken();
+        if (getBranch().getForkingMarket() == this) {
+            return true;
+        }
+        if (address(automatedReporterDisputeBondToken) != NULL_ADDRESS && automatedReporterDisputeBondToken.getDisputedPayoutDistributionHash() == finalPayoutDistributionHash) {
+            _reputationToken.trustedTransfer(automatedReporterDisputeBondToken, getFinalWinningReportingToken(), _reputationToken.balanceOf(automatedReporterDisputeBondToken));
+        }
+        if (address(limitedReportersDisputeBondToken) != NULL_ADDRESS && limitedReportersDisputeBondToken.getDisputedPayoutDistributionHash() == finalPayoutDistributionHash) {
+            _reputationToken.trustedTransfer(limitedReportersDisputeBondToken, getFinalWinningReportingToken(), _reputationToken.balanceOf(limitedReportersDisputeBondToken));
+        }
+        return true;
+    }
+
+    function derivePayoutDistributionHash(uint256[] _payoutNumerators) public constant returns (bytes32) {
+        uint256 _sum = 0;
+        require(_payoutNumerators.length == numOutcomes);
```

Comment:
I think we can delete this require, since the `require(_sum == payoutDenominator)` means no individual payout numerator can be greater than payout denominator.

This was needed previously to prevent someone including a negative number in _payoutNumerators, but now we have unsigned types for that.  If you do remove this, we should change the sum to `_sum = _sum.add(_payoutNumerators[i]);` to prevent overflows dodging the check.  We should generally be using the safe math function here anyway rather than trusting ourselves to correctly prevent overflow with require statements.